x-pack/plugins/ml/public/application/timeseriesexplorer/components/timeseries_chart/timeseries_chart_with_tooltip.tsx

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useEffect, useState, useCallback } from 'react';
+import { i18n } from '@kbn/i18n';
+import { MlTooltipComponent } from '../../../components/chart_tooltip';
+import { TimeseriesChart } from './timeseries_chart';
+import { CombinedJob } from '../../../../../common/types/anomaly_detection_jobs';
+import { ml } from '../../../services/ml_api_service';
+import { ANNOTATIONS_TABLE_DEFAULT_QUERY_SIZE } from '../../../../../common/constants/search';
+import { extractErrorMessage } from '../../../../../common/util/errors';
+import { Annotation } from '../../../../../common/types/annotations';
+import { useNotifications } from '../../../contexts/kibana';
+
+interface TimeSeriesChartWithTooltipsProps {
+  bounds: any;
+  selectedDetectorIndex: number;
+  renderFocusChartOnly: boolean;
+  selectedJob: CombinedJob;
+  showAnnotations: boolean;
+  showForecast: boolean;
+  showModelBounds: boolean;
+  chartProps: any;
+  lastRefresh: number;
+}
+export const TimeSeriesChartWithTooltips: FC<TimeSeriesChartWithTooltipsProps> = ({
+  bounds,
+  selectedDetectorIndex,
+  renderFocusChartOnly,
+  selectedJob,
+  showAnnotations,
+  showForecast,
+  showModelBounds,
+  chartProps,
+  lastRefresh,
+}) => {
+  const { toasts: toastNotifications } = useNotifications();
+
+  const [annotationData, setAnnotationData] = useState<Annotation[]>([]);
+
+  const showAnnotationErrorToastNotification = useCallback((error?: string) => {
+    toastNotifications.addDanger({
+      title: i18n.translate(
+        'xpack.ml.timeSeriesExplorer.mlSingleMetricViewerChart.annotationsErrorTitle',
+        {
+          defaultMessage: 'An error occurred fetching annotations',
+        }
+      ),
+      ...(error ? { text: extractErrorMessage(error) } : {}),
+    });
+  }, []);
+
+  useEffect(() => {
+    let unmounted = false;
+    /**
+     * Loads the full list of annotations for job without any aggs or time boundaries
+     * used to indicate existence of annotations that are beyond the selected time
+     * in the time series brush area
+     */
+    const loadAnnotations = async (jobId: string) => {
+      try {
+        const resp = await ml.annotations.getAnnotations({
+          jobIds: [jobId],
+          earliestMs: null,
```

Comment:
I think the time range bounds for this query needs to be set to the bounds of the time picker, so it only loads annotations which are within the time range of the page. I don't think we should be showing annotations which are outside the time range of the page:

![image](https://user-images.githubusercontent.com/7405507/97979984-fde0d600-1dc7-11eb-97ba-2caa7f0dc9d0.png)

Note that the earliest / latest of the query need to set in such a way that it loads annotations whose range is partly inside the time range of the view.

![image](https://user-images.githubusercontent.com/7405507/97980344-8790a380-1dc8-11eb-8313-d77114fe8dc5.png)

Reply:
Updated to load only annotations which are within the time range of the page here https://github.com/elastic/kibana/pull/81490/commits/3fb8c08929d8209b880f27b05db556e6a21c195b